File: ruby/lib/steam/packets/a2s_serverquery_getchallenge_packet.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2008-2011, Sebastian Staudt

require 'steam/packets/steam_packet'

# This packet class represents a A2S_SERVERQUERY_GETCHALLENGE request send to
# a game server
#
# It is used to retrieve a challenge number from the game server, which helps
# to identify the requesting client.
#
# @author Sebastian Staudt
# @see GameServer#update_challenge_number
class A2S_SERVERQUERY_GETCHALLENGE_Packet

  include SteamPacket

  # Creates a new A2S_SERVERQUERY_GETCHALLENGE request object
  def initialize
    super A2S_SERVERQUERY_GETCHALLENGE_HEADER
  end

end

File: php/lib/steam/packets/A2S_SERVERQUERY_GETCHALLENGE_Packet.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Packets
 */

require_once "steam/packets/SteamPacket.php";

/**
 * @package Steam Condenser (PHP)
 * @subpackage Packets
 */
class A2S_SERVERQUERY_GETCHALLENGE_Packet extends SteamPacket
{
	/**
	 *
	 */
	public function __construct()
	{
		parent::__construct(SteamPacket::A2S_SERVERQUERY_GETCHALLENGE_HEADER);
	}
}
?>